### PR TITLE
Separate function for future reusing

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- base64 decode function is extracted for reuse
- applied padding auto augmentation for base64 JWT which may be without padding
- ASN1 size check applied for small size case

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

**Review by Yullia — APPROVED** [review]

File: credsweeper/utils/util.py [hunk 1]
```diff
@@ -613,3 +616,17 @@ def parse_python(source: str) -> List[Any]:
...
+
+    @staticmethod
+    def decode_base64(text: str, padding_safe: bool = False, urlsafe_detect=False) -> bytes:
+        """decode text to bytes with / without padding detect and urlsafe symbols"""
+        value = text
+        if padding_safe:
+            pad_num = 0x3 & len(value)
+            if pad_num:
+                value += '=' * (4 - pad_num)
+        if urlsafe_detect and '-' in value or '_' in value:
```
Comment (kmnls) [inline credsweeper/utils/util.py:628]:
```suggestion
        if urlsafe_detect and ('-' in value or '_' in value):
```
AND has higher precedence then OR

**Review by kmnls — CHANGES_REQUESTED** [review]
See comment

**Comment by babenek** [comment]
> See comment

Thank you for review. The issue was fixed.

**Review by xDizzix — APPROVED** [review]
+

**Review by kmnls — APPROVED** [review]
Agree

**Pull request merged**